src/main/java/org/jabref/gui/copyfiles/CopySingleFileAction.java

```diff
@@ -0,0 +1,60 @@
+package org.jabref.gui.copyfiles;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import org.jabref.Globals;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.util.DirectoryDialogConfiguration;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.util.OptionalUtil;
+import org.jabref.preferences.JabRefPreferences;
+
+public class CopySingleFileAction {
+
+    private LinkedFile linkedFile;
+    private DialogService dialogService;
+    private BibDatabaseContext databaseContext;
+    private final BiFunction<Path, Path, Path> resolvePathFilename = (path, file) -> {
+        return path.resolve(file.getFileName());
+    };
+
+    public CopySingleFileAction(LinkedFile linkedFile, DialogService dialogService, BibDatabaseContext databaseContext) {
+        this.linkedFile = linkedFile;
+        this.dialogService = dialogService;
+        this.databaseContext = databaseContext;
+    }
+
+    public void copyFile() {
+        DirectoryDialogConfiguration dirDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                                                                                                        .withInitialDirectory(Paths.get(Globals.prefs.get(JabRefPreferences.EXPORT_WORKING_DIRECTORY)))
+                                                                                                        .build();
+        Optional<Path> exportPath = dialogService.showDirectorySelectionDialog(dirDialogConfiguration);
+        exportPath.ifPresent(this::copyFileToDestination);
+
+    }
+
+    private void copyFileToDestination(Path exportPath) {
+        Optional<Path> fileToExport = linkedFile.findIn(databaseContext, Globals.prefs.getFilePreferences());
+
+        Optional<Path> newPath = OptionalUtil.combine(Optional.of(exportPath), fileToExport, resolvePathFilename);
```

Comment:
I think this is a bit overly complicated as we can only copy the file if we find it (e.g. fileToExport is not empty). Moreover, please show a warning in case it is empty.